wandb/sdk/service/service.py

```diff
@@ -3,79 +3,57 @@
 Reliably launch and connect to grpc process.
 """
 
-import datetime
-import enum
-import logging
 import os
 import subprocess
 import sys
 import tempfile
 import time
 from typing import Any, Dict, Optional
-from typing import TYPE_CHECKING
-
-import grpc
-from wandb.proto import wandb_server_pb2 as spb
-from wandb.proto import wandb_server_pb2_grpc as pbgrpc
-from wandb.sdk.wandb_settings import Settings
-
-if TYPE_CHECKING:
-    from google.protobuf.internal.containers import MessageMap
-
-
-def _pbmap_apply_dict(
-    m: "MessageMap[str, spb.SettingsValue]", d: Dict[str, Any]
-) -> None:
-    for k, v in d.items():
-        if isinstance(v, datetime.datetime):
-            continue
-        if isinstance(v, enum.Enum):
-            continue
-        sv = spb.SettingsValue()
-        if v is None:
-            sv.null_value = True
-        elif isinstance(v, int):
-            sv.int_value = v
-        elif isinstance(v, float):
-            sv.float_value = v
-        elif isinstance(v, str):
-            sv.string_value = v
-        elif isinstance(v, bool):
-            sv.bool_value = v
-        elif isinstance(v, tuple):
-            sv.tuple_value.string_values.extend(v)
-        m[k].CopyFrom(sv)
+
+from . import port_file
+from .service_base import ServiceInterface
+from .service_sock import ServiceSockInterface
+
+# from .service_grpc import ServiceGrpcInterface
 
 
 class _Service:
-    _stub: Optional[pbgrpc.InternalServiceStub]
+    _grpc_port: Optional[int]
+    _sock_port: Optional[int]
+    _service_interface: ServiceInterface
+    _internal_proc: Optional[subprocess.Popen]
 
     def __init__(self) -> None:
         self._stub = None
+        self._grpc_port = None
+        self._sock_port = None
+        self._service_interface = ServiceSockInterface()
```

Comment:
should this one be a `ServiceInterface`?

Reply:
yes, but im still working on that change to bring back grpc, coming in the next hour 🤞

Reply:
It is now (ugly but more correct):
```
        if _use_grpc:
            from .service_grpc import ServiceGrpcInterface

            self._service_interface = ServiceGrpcInterface()
        else:
            self._service_interface = ServiceSockInterface()
```